pkg/naming/name_generator_test.go

```diff
@@ -0,0 +1,51 @@
+package naming_test
+
+import (
+	"math/rand"
+	"strings"
+
+	"github.com/aslakknutsen/istio-workspace/pkg/naming"
+	. "github.com/aslakknutsen/istio-workspace/test/matchers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Name generation (used for k8s objects such as namespaces, sessions etc)", func() {
+
+	It("should always generate lowercase string", func() {
+		name := naming.RandName(32)
+		Expect(name).To(Equal(strings.ToLower(name)))
+	})
+
+	It("should generate empty string when 0 length requested", func() {
+		name := naming.RandName(0)
+		Expect(name).To(BeEmpty())
+	})
+
+	It("should always generate letter when single character name is requested", func() {
+		name := naming.RandName(1)
+		Expect(name).To(OnlyContain("abcdefghijklmnopqrstuvwxyz"))
+	})
+
+	It("should generate name with first character being a letter", func() {
+		name := naming.RandName(32)
+		Expect(name).To(StartWithLetter())
```

Comment:
these are a bit random if they fail or not are they not?

Reply:
If there's a bug - yes. But there is not :)

As I said in the PR description - ideally it should be done using generators, but I would rather first ship that and then use generators to improve tests.

Reply:
Why not just remove '-' from the options. It's not like it's required :)

Reply:
Still, cannot be a number

Reply:
We could simplify though and remove those tests, true. You favor this over what we have now?

Reply:
we don't need to use numbers either.. :)

Reply:
ACK

Reply:
Fixed with 862840b. I just wanted it to look fancy :P